pkg/apis/datadoghq/v1alpha1/datadogagent_default.go

```diff
@@ -402,10 +397,6 @@ func DefaultDatadogAgentSpecAgentConfig(config *NodeAgentConfig) *NodeAgentConfi
 		config = &NodeAgentConfig{}
 	}
 
-	if config.DDUrl == nil {
-		config.DDUrl = NewStringPointer(defaultDatadogURL)
```

Comment:
Do not try to assign a value (in this case, default value) for `config.DDUrl` if it is not set